[TensorPipe] Give higher priority to CPU-only channels.

beauby opened the pull request:
Summary:
CUDA channels might implement CPU-to-CPU transfers, but will usually be
less efficient for that purpose.

Test Plan: CI

Differential Revision: D27994069

lw reviewed (commented):
I'm not sure I get this change, the priorities between CPU and CUDA channels still seem quite "interleaved", e.g., from low to high I see this sequence: CUDA basic -> basic -> MPT UV -> CUDA GDR -> CMA. Should we instead restrict the two groups to entirely disjoint ranges? E.g., use range 0-999 for CUDA, and range 1000+ for CPU?

Commit:
[TensorPipe] Give higher priority to CPU-only channels. (#56908)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56908

CUDA channels might implement CPU-to-CPU transfers, but will usually be
less efficient for that purpose.

Test Plan: CI

Differential Revision: D27994069

fbshipit-source-id: d418e12ca4758dc2da2ad85c64a37ab17d34c6b1

lw reviewed (approved):
Looks good! The only thing is that perhaps we could add a comment explaining our reasoning, for future readers?
